pallets/rmrk-core/src/lib.rs

```diff
@@ -403,10 +422,32 @@ pub mod pallet {
 				new_owner.clone(),
 				max_recursions,
 			)?;
+			// Check if root owner is a nft & call pallet_uniques::do_transfer w/ actual account owner
+			match Pallet::<T>::decode_nft_account_id::<T::AccountId>(new_owner_account.clone()) {
+				None => {
+					pallet_uniques::Pallet::<T>::do_transfer(
+						collection_id,
+						nft_id,
+						new_owner_account.clone(),
+						|class_details, details|
+							Ok(())
+					)?;
+				},
+				Some((cid, nid)) => {
+					let (dest, _) = Pallet::<T>::lookup_root_owner(cid, nid)?;
```

Comment:
The destination shouldn't be the the root owner of the target. At the uniques level, we should transfer the current NFT to the virtual account of the parent NFT (new_owenr_account).